src/uipath/_cli/_evals/mocks/mocks.py

```
@@ -0,0 +1,140 @@
+"""Mocking interface."""
+
+import asyncio
+import functools
+import inspect
+import logging
+import threading
+from contextvars import ContextVar
+from typing import Any, Callable, Optional
+
+from pydantic import BaseModel, TypeAdapter
+from pydantic_function_models import ValidatedFunction  # type: ignore[import-untyped]
+
+from uipath._cli._evals._models._evaluation_set import EvaluationItem
+from uipath._cli._evals.mocks.mocker import Mocker
+from uipath._cli._evals.mocks.mocker_factory import MockerFactory
+
+evaluation_context: ContextVar[Optional[EvaluationItem]] = ContextVar(
+    "evaluation", default=None
+)
+
+mocker_context: ContextVar[Optional[Mocker]] = ContextVar("mocker", default=None)
+
+
+def set_evaluation_item(item: EvaluationItem) -> None:
+    """Set an evaluation item within an evaluation set."""
+    evaluation_context.set(item)
+    try:
+        mocker_context.set(MockerFactory.create(item))
+    except Exception:
+        logger.warning(f"Failed to create mocker for evaluation {item.name}")
+        mocker_context.set(None)
+
+
+async def get_mocked_response(
+    func: Callable[[Any], Any], params: dict[str, Any], *args, **kwargs
+) -> Any:
+    """Get a mocked response."""
+    mocker = mocker_context.get()
+    evaluation_item = evaluation_context.get()
+    if mocker is None or evaluation_item is None:
+        # TODO raise a new UiPath exception type
+        raise RuntimeError(f"Evaluation item {func.__name__} has not been evaluated")
+    else:
+        return await mocker.response(func, params, *args, **kwargs)
+
+
+_event_loop = None
+logger = logging.getLogger(__name__)
+
+
+def run_coroutine(coro):
+    """Run a coroutine synchronously."""
+    global _event_loop
```

why do we need a persistent loop here? if the loop dies won't the subsequent calls fail silently?